File: frontend/src/types/extractionProgress.ts
export interface ExtractionSubmitResponse {
  extraction_id: string;
}

export interface ExtractionProgressResult {
  project_id: string;
  source_corpus: string;
  archetypal_patterns: number;
  narrative_structures: number;
  world_rules: number;
  symbolic_motifs: number;
  warnings: string[];
}

export interface ExtractionProgressResponse {
  extraction_id: string;
  status: 'pending' | 'running' | 'completed' | 'failed';
  phase: string;
  result: ExtractionProgressResult | null;
  error: string | null;
}

File: app/schemas/extraction_progress.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ExtractionProgressResult(BaseModel):
    """Result returned when extraction completes."""
    project_id: str = ""
    source_corpus: str = ""
    archetypal_patterns: int = 0
    narrative_structures: int = 0
    world_rules: int = 0
    symbolic_motifs: int = 0
    warnings: list[str] = Field(default_factory=list)


class ExtractionProgressResponse(BaseModel):
    """Response for polling extraction job status."""
    extraction_id: str
    status: str  # pending | running | completed | failed
    phase: str = ""
    result: ExtractionProgressResult | None = None
    error: str | None = None


class ExtractionSubmitResponse(BaseModel):
    """Response when submitting an async extraction job."""
    extraction_id: str
